File: 01/0105js/최대공약수와_최소공배수.js
// function gcd(n, m) {
//   return n % m === 0 ? m : gcd(m, n % m);
// }

// function solution(n, m) {
//   const _gcd = gcd(n, m);
//   return [_gcd, (n * m) / _gcd];
// }
function solution(a, b) {
  var r;
  for (var ab = a * b; (r = a % b); a = b, b = r) {}
  return [b, ab / b];
}
console.log(solution(3, 5));

File: 백준/파이썬/기초1/수학/최대공약수와 최소공배수.py
import sys
sys.stdin = open('2609.txt', 'r')
input = sys.stdin.readline
A, B = map(int, input().split())


def gcd(A, B):
    if B == 0:
        return A
    else:
        return gcd(B, A % B)


g = gcd(A, B)
print(g)
print(g * (A // g) * (B // g))
